src/frameworks/nuxt/index.ts

```diff
@@ -31,47 +41,119 @@ export async function discover(dir: string): Promise<{ mayWantBackend: true } |
   const anyConfigFileExists = await nuxtConfigFilesExist(dir);
 
   if (!anyConfigFileExists && !nuxtDependency) return;
-  if (version && gte(version, "3.0.0-0")) return { mayWantBackend: true };
+  if (version && gte(version, "3.0.0-0")) return { mayWantBackend: true, publicDirectory: "" };
 
   return;
 }
 
+/**
+ *
+ */
 export async function build(root: string) {
-  const { buildNuxt } = await relativeRequire(root, "@nuxt/kit");
-  const nuxtApp = await getNuxt3App(root);
+  overrideEnv("production");
+  const { buildNuxt, loadNuxt, useNitro } = await relativeRequire(root, "@nuxt/kit");
 
   await warnIfCustomBuildScript(root, name, DEFAULT_BUILD_SCRIPT);
 
-  await buildNuxt(nuxtApp);
-  return { wantsBackend: true };
-}
+  /**
+   * currently genrate is only supported by using nuxi generate or nuxi build --prerender
+   * moreover, it is currently still experimental
+   */
+  const prerender = false;
```

Comment:
We can turn this into a CLI experiment

Reply:
Prerender is one of several ways to build a nuxt app. This way the whole app is generated locally and no backend is needed. Other build options such as spa or ssr can be configured via config for individual routes, for which prerender is then set to false.
To use prerender as a whole app, you would need to use a cli argument, such as "firebase deploy --nuxt-preprender" or there is a way to specify prerender as config via firebase.json.

Reply:
Ah gotcha, I'll have to think about that more then... 

I've been wondering if we should just shell out to `npm run build` in all these integrations and change the default build step warning to something like `we noticed your build step is not just "${default}". Here be dragons—do what you want—we expect the compiled artifacts to be in ${distDir} at the end of the day`. That way if they want prerender they just change the build step in their package.json. 

Doing all the building in Node has theoretical benefits... but it seems the cost of maintaining these build steps and adding tons of framework specific options outweighs those. As an external contributor to our framework integrations do you agree or have any other feedback about our choice of abstraction?

Reply:
Yes, that makes perfect sense, as it would really be more maintenance free.
But of course it would also be possible to throw out the config for prerender, because the user can really overwrite it in the nuxt.config. The `_generate` config only affects a complete static build of the page anyway. Building the app with different strategies is always possible on route level too. 
Furthermore, with a build script it is no longer so easy to see if the nuxt app really needs a backend or not, since you don't have access to the configuration of nuxt.
I would of course find building the app via node nicer and also more practical but could of course understand the decision.

Reply:
Maybe I can implement both things. If the user defines a build script, then this is used. If there is no build script, the implemented node builder is used.
Here I would take out prerender and read via the generated build config of loadNuxt if a backend is needed or not.

Reply:
Let me and the team think on this strategy a bit before we reshuffle, in the meantime we'll review the rest of this PR.

Reply:
@DevJoghurt Is it possible to detect the `prerender` config from a user's config/project setup?

cc @jamesdaniels